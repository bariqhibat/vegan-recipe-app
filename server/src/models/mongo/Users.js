import mon, { Schema } from "mongoose";
import beautifyUnique from "mongoose-beautiful-unique-validation";

const Users = Schema({
  firstName: {
    type: String,
    required: [true, "First name is required"],
    lowercase: true,
    trim: true,
  },
  lastName: {
    type: String,
    required: [true, "Last name is required"],
    lowercase: true,
    trim: true,
  },
  email: {
    type: String,
    required: [true, "Email is required"],
    unique: "Email already exists",
    lowercase: true,
    trim: true,
    index: true,
    match: [
      // eslint-disable-next-line
      /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/,
      "Invalid email format",
    ],
  },
  password: {
    type: String,
    minlength: [5, "Password must be longer than 5 chars"],
    maxlength: [64, "Password must be shorter than 65 chars"],
  },
});

// Enable beautifying on this schema
Users.plugin(beautifyUnique);

export default mon.model("users", Users);
