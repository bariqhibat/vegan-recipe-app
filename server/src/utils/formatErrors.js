import mon from "mongoose";
import { pick } from "lodash";

const formatErrors = (e) => {
  if (e instanceof mon.Error.ValidationError) {
    return Object.keys(e.errors).map((keyName) => {
      return pick(e.errors[keyName].properties, ["path", "message"]);
    });
  }

  return [{ path: "name", message: "something went wrong" }];
};

export default formatErrors;
