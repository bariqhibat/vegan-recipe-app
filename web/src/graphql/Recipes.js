import { gql } from "@apollo/client";

export const CREATE_RECIPE = gql`
  mutation(
    $title: String!
    $image: String
    $ingredients: [Ingredient_Input!]!
    $instructions: [String!]!
    $userUUID: String
  ) {
    createRecipe(
      title: $title
      image: $image
      ingredients: $ingredients
      instructions: $instructions
      userUUID: $userUUID
    ) {
      ok
      errors {
        path
        message
      }
      recipe {
        _id
        title
        image
        instructions
        userUUID {
          _id
        }
        ingredients {
          name
          amount
        }
      }
    }
  }
`;

export const UPDATE_RECIPE_BY_ID = gql`
  mutation(
    $recipeUUID: String!
    $title: String
    $image: String
    $ingredients: [Ingredient_Input!]
    $instructions: [String!]
  ) {
    updateRecipeById(
      recipeUUID: $recipeUUID
      title: $title
      image: $image
      ingredients: $ingredients
      instructions: $instructions
    ) {
      ok
      errors {
        path
        message
      }
      recipe {
        _id
        title
        image
        instructions
        userUUID {
          _id
        }
        ingredients {
          name
          amount
        }
      }
    }
  }
`;

export const DELETE_RECIPE_BY_ID = gql`
  mutation($recipeUUID: String!) {
    deleteRecipeById(recipeUUID: $recipeUUID) {
      ok
      errors {
        path
        message
      }
      recipe {
        _id
        title
        image
        instructions
        userUUID {
          _id
        }
        ingredients {
          name
          amount
        }
      }
    }
  }
`;

export const GET_ALL_RECIPES = gql`
  query {
    getAllRecipes {
      ok
      errors {
        path
        message
      }
      recipes {
        _id
        title
        image
        instructions
        userUUID {
          _id
        }
        ingredients {
          name
          amount
        }
      }
    }
  }
`;

export const GET_RECIPE_BY_ID = gql`
  query($recipeUUID: String!) {
    getRecipeById(recipeUUID: $recipeUUID) {
      ok
      errors {
        path
        message
      }
      recipe {
        _id
        title
        image
        userUUID {
          _id
          firstName
        }
        instructions
        ingredients {
          name
          amount
        }
      }
    }
  }
`;
export const GET_RECIPES_BY_USER_ID = gql`
  query($userUUID: String) {
    getRecipeByUserUUID(userUUID: $userUUID) {
      ok
      errors {
        path
        message
      }
      recipes {
        _id
        title
        image
        instructions
        userUUID {
          _id
        }
        ingredients {
          name
          amount
        }
      }
    }
  }
`;
