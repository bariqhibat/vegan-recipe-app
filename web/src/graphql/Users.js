import { gql } from "@apollo/client";

export const EMAIL_LOGIN = gql`
  mutation($email: String!, $password: String!) {
    login(email: $email, password: $password) {
      ok
      errors {
        path
        message
      }
      token
      refreshToken
      user {
        _id
      }
    }
  }
`;
export const EMAIL_SIGNUP = gql`
  mutation(
    $firstName: String!
    $lastName: String!
    $email: String!
    $password: String!
  ) {
    createUser(
      firstName: $firstName
      lastName: $lastName
      email: $email
      password: $password
    ) {
      ok
      errors {
        path
        message
      }
      user {
        _id
        email
      }
    }
  }
`;

export const ME_QUERY = gql`
  query($userUUID: String) {
    meQuery(userUUID: $userUUID) {
      ok
      errors {
        path
        message
      }
      user {
        _id
        firstName
        lastName
        email
      }
    }
  }
`;
