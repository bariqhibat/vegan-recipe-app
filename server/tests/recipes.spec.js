import {
  CREATE_RECIPE,
  UPDATE_RECIPE_BY_ID,
  DELETE_RECIPE_BY_ID,
  GET_RECIPES_BY_USER_ID,
  GET_ALL_RECIPES,
  GET_RECIPE_BY_ID,
} from "../../web/src/graphql/Recipes";
import { EMAIL_SIGNUP } from "../../web/src/graphql/Users";
import { mutate, query } from "./utils/providers";

let data = {
  user: {
    _id: null,
    firstName: "Bariq",
    lastName: "Nurlis",
    email: "bariqnurlis@bariq.com",
    password: "lklklklk",
  },
  recipe: {
    title: "pepe",
    image:
      "https://static01.nyt.com/images/2016/09/28/us/17xp-pepethefrog_web1/28xp-pepefrog-articleLarge.jpg?quality=75&auto=webp&disable=upscale",
    ingredients: [
      {
        name: "carrot",
        amount: 2,
      },
      {
        name: "oregano",
        amount: 10,
      },
      {
        name: "pickles",
        amount: 2,
      },
    ],
    instructions: [
      "this is a test",
      "this is a test",
      "this is a test",
      "this is a test",
      "this is a test",
      "this is a test",
    ],
  },
};

describe("Query", () => {
  // ! Environment setup
  let recipeUUID = null;
  beforeEach(async () => {
    const res = await mutate(CREATE_RECIPE, {
      variables: {
        ...data.recipe,
      },
    });

    recipeUUID = res.data.createRecipe.recipe._id;
  });

  describe("getRecipeById", () => {
    describe("Successful", () => {
      test("Successful", async () => {
        const res = await query(GET_RECIPE_BY_ID, {
          variables: {
            recipeUUID,
          },
        });

        expect(res).toMatchObject({
          data: {
            getRecipeById: {
              ok: true,
              errors: null,
              recipe: {
                _id: expect.any(String),
                ...data.recipe,
              },
            },
          },
        });
      });
    });
    describe("Unsuccessful", () => {
      test("Missing args", async () => {
        await expect(async () => {
          await mutate(GET_RECIPE_BY_ID).catch((err) => {
            throw err;
          });
        }).rejects.toThrowErrorMatchingSnapshot();
      });
      describe("Wrong type of args", () => {
        it.each`
          email
          ${""}
          ${3123}
          ${[3123, 312]}
          ${["asdasd", "dasda"]}
          ${null}
          ${undefined}
        `("'$recipeUUID' of recipeUUID variant", async ({ recipeUUID }) => {
          if (recipeUUID !== null && recipeUUID !== undefined) {
            const res = await mutate(CREATE_RECIPE, {
              variables: {
                recipeUUID,
              },
            });

            expect(res).toMatchSnapshot();
            expect(res).toMatchObject({
              data: {
                getRecipeById: {
                  ok: false,
                  recipe: null,
                },
              },
            });
          } else {
            await expect(async () => {
              await mutate(CREATE_RECIPE, {
                variables: {
                  recipeUUID,
                },
              }).catch((err) => {
                throw err;
              });
            }).rejects.toThrowErrorMatchingSnapshot();
          }
        });
      });
    });
  });
  describe("getRecipeByUserUUID", () => {
    let userUUID = null;
    beforeEach(async () => {
      const {
        data: {
          createUser: {
            user: { _id },
          },
        },
      } = await mutate(EMAIL_SIGNUP, {
        variables: {
          ...data.user,
        },
      });

      userUUID = _id;

      await mutate(CREATE_RECIPE, {
        variables: {
          ...data.recipe,
          userUUID,
        },
      });
    });

    describe("Successful", () => {
      test("Successful", async () => {
        const res = await query(GET_RECIPES_BY_USER_ID, {
          variables: {
            userUUID,
          },
        });

        expect(res).toMatchObject({
          data: {
            getRecipeByUserUUID: {
              ok: true,
              recipes: [
                {
                  _id: expect.any(String),
                  ...data.recipe,
                },
              ],
            },
          },
        });
      });
    });
    describe("Unsuccessful", () => {});
  });
  describe("getAllRecipes", () => {
    describe("Successful", () => {
      test("Successful", async () => {
        const res = await query(GET_ALL_RECIPES);

        expect(res).toMatchObject({
          data: {
            getAllRecipes: {
              ok: true,
              recipes: [
                {
                  _id: expect.any(String),
                  ...data.recipe,
                },
              ],
            },
          },
        });
      });
    });
    describe("Unsuccessful", () => {});
  });
});
describe("Mutation", () => {
  describe("createRecipe", () => {
    describe("Successful", () => {
      test("Non-user", async () => {
        const res = await mutate(CREATE_RECIPE, {
          variables: {
            ...data.recipe,
          },
        });

        expect(res).toMatchObject({
          data: {
            createRecipe: {
              ok: true,
              recipe: {
                _id: expect.any(String),
                ...data.recipe,
              },
            },
          },
        });
      });
      // ! Unable since requires express middleware
      test("User", async () => {
        const {
          data: {
            createUser: {
              user: { _id: userUUID },
            },
          },
        } = await mutate(EMAIL_SIGNUP, {
          variables: {
            ...data.user,
          },
        });

        const res = await mutate(CREATE_RECIPE, {
          variables: {
            ...data.recipe,
            userUUID,
          },
        });

        expect(res).toMatchObject({
          data: {
            createRecipe: {
              ok: true,
              recipe: {
                _id: expect.any(String),
                ...data.recipe,
              },
            },
          },
        });
      });
    });
    describe("Unsuccessful", () => {
      test("Missing args", async () => {
        await expect(async () => {
          await mutate(CREATE_RECIPE).catch((err) => {
            throw err;
          });
        }).rejects.toThrowErrorMatchingSnapshot();
      });
      describe("Wrong type of args", () => {
        it.each`
          email
          ${""}
          ${"bobby"}
          ${3123}
          ${["asdasd", "dasda"]}
          ${null}
          ${undefined}
        `("'$ingredients' of ingredients variant", async ({ ingredients }) => {
          if (typeof ingredients === "string") {
            const res = await mutate(CREATE_RECIPE, {
              variables: {
                ...data.recipe,
                ingredients,
              },
            });

            expect(res).toMatchSnapshot();
            expect(res).toMatchObject({
              data: {
                createUser: {
                  ok: false,
                  user: null,
                },
              },
            });
          } else {
            await expect(async () => {
              await mutate(CREATE_RECIPE, {
                variables: {
                  ...data.user,
                  ingredients,
                },
              }).catch((err) => {
                throw err;
              });
            }).rejects.toThrowErrorMatchingSnapshot();
          }
        });
      });
    });
  });
  describe("updateRecipeById", () => {
    let recipeUUID = null;
    beforeEach(async () => {
      const res = await mutate(CREATE_RECIPE, {
        variables: {
          ...data.recipe,
        },
      });

      recipeUUID = res.data.createRecipe.recipe._id;
    });

    describe("Successful", () => {
      test("Successful", async () => {
        const res = await mutate(UPDATE_RECIPE_BY_ID, {
          variables: {
            ...data.recipe,
            title: "Changed title",
            recipeUUID,
          },
        });

        expect(res).toMatchObject({
          data: {
            updateRecipeById: {
              ok: true,
              errors: null,
              recipe: {
                _id: expect.any(String),
                title: "Changed title",
                ...data.recipe,
              },
            },
          },
        });
      });
    });
    describe("Unsuccessful", () => {
      test("Missing args", async () => {
        await expect(async () => {
          await mutate(UPDATE_RECIPE_BY_ID).catch((err) => {
            throw err;
          });
        }).rejects.toThrowErrorMatchingSnapshot();
      });
    });
  });
  describe("deleteRecipeById", () => {
    let recipeUUID = null;
    beforeEach(async () => {
      const res = await mutate(CREATE_RECIPE, {
        variables: {
          ...data.recipe,
        },
      });

      recipeUUID = res.data.createRecipe.recipe._id;
    });

    describe("Successful", () => {
      test("Successful", async () => {
        const res = await mutate(DELETE_RECIPE_BY_ID, {
          variables: {
            recipeUUID,
          },
        });

        expect(res).toMatchObject({
          data: {
            deleteRecipeById: {
              ok: true,
              errors: null,
              recipe: {
                _id: expect.any(String),
                ...data.recipe,
              },
            },
          },
        });
      });
    });
    describe("Unsuccessful", () => {
      test("Missing recipe", async () => {
        await mutate(DELETE_RECIPE_BY_ID, {
          variables: {
            recipeUUID,
          },
        });

        const res = await mutate(DELETE_RECIPE_BY_ID, {
          variables: {
            recipeUUID,
          },
        });

        expect(res).toMatchSnapshot();
      });
      test("Missing args", async () => {
        await expect(async () => {
          await mutate(DELETE_RECIPE_BY_ID).catch((err) => {
            throw err;
          });
        }).rejects.toThrowErrorMatchingSnapshot();
      });
    });
  });
});
