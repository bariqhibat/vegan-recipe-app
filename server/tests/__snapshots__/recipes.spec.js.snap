// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mutation createRecipe Unsuccessful Missing args 1`] = `
"{\\"errors\\":[{\\"message\\":\\"Variable \\\\\\"$title\\\\\\" of required type \\\\\\"String!\\\\\\" was not provided.\\",\\"locations\\":[{\\"line\\":1,\\"column\\":11}],\\"extensions\\":{\\"code\\":\\"INTERNAL_SERVER_ERROR\\"}},{\\"message\\":\\"Variable \\\\\\"$ingredients\\\\\\" of required type \\\\\\"[Ingredient_Input!]!\\\\\\" was not provided.\\",\\"locations\\":[{\\"line\\":1,\\"column\\":44}],\\"extensions\\":{\\"code\\":\\"INTERNAL_SERVER_ERROR\\"}},{\\"message\\":\\"Variable \\\\\\"$instructions\\\\\\" of required type \\\\\\"[String!]!\\\\\\" was not provided.\\",\\"locations\\":[{\\"line\\":1,\\"column\\":80}],\\"extensions\\":{\\"code\\":\\"INTERNAL_SERVER_ERROR\\"}}]}
"
`;

exports[`Mutation createRecipe Unsuccessful Wrong type of args '$ingredients' of ingredients variant 1`] = `
"{\\"errors\\":[{\\"message\\":\\"Variable \\\\\\"$title\\\\\\" of required type \\\\\\"String!\\\\\\" was not provided.\\",\\"locations\\":[{\\"line\\":1,\\"column\\":11}],\\"extensions\\":{\\"code\\":\\"INTERNAL_SERVER_ERROR\\"}},{\\"message\\":\\"Variable \\\\\\"$ingredients\\\\\\" got invalid value undefined; Expected non-nullable type \\\\\\"[Ingredient_Input!]!\\\\\\" not to be null.\\",\\"locations\\":[{\\"line\\":1,\\"column\\":44}],\\"extensions\\":{\\"code\\":\\"INTERNAL_SERVER_ERROR\\"}},{\\"message\\":\\"Variable \\\\\\"$instructions\\\\\\" of required type \\\\\\"[String!]!\\\\\\" was not provided.\\",\\"locations\\":[{\\"line\\":1,\\"column\\":80}],\\"extensions\\":{\\"code\\":\\"INTERNAL_SERVER_ERROR\\"}}]}
"
`;

exports[`Mutation createRecipe Unsuccessful Wrong type of args '$ingredients' of ingredients variant 2`] = `
"{\\"errors\\":[{\\"message\\":\\"Variable \\\\\\"$title\\\\\\" of required type \\\\\\"String!\\\\\\" was not provided.\\",\\"locations\\":[{\\"line\\":1,\\"column\\":11}],\\"extensions\\":{\\"code\\":\\"INTERNAL_SERVER_ERROR\\"}},{\\"message\\":\\"Variable \\\\\\"$ingredients\\\\\\" got invalid value undefined; Expected non-nullable type \\\\\\"[Ingredient_Input!]!\\\\\\" not to be null.\\",\\"locations\\":[{\\"line\\":1,\\"column\\":44}],\\"extensions\\":{\\"code\\":\\"INTERNAL_SERVER_ERROR\\"}},{\\"message\\":\\"Variable \\\\\\"$instructions\\\\\\" of required type \\\\\\"[String!]!\\\\\\" was not provided.\\",\\"locations\\":[{\\"line\\":1,\\"column\\":80}],\\"extensions\\":{\\"code\\":\\"INTERNAL_SERVER_ERROR\\"}}]}
"
`;

exports[`Mutation createRecipe Unsuccessful Wrong type of args '$ingredients' of ingredients variant 3`] = `
"{\\"errors\\":[{\\"message\\":\\"Variable \\\\\\"$title\\\\\\" of required type \\\\\\"String!\\\\\\" was not provided.\\",\\"locations\\":[{\\"line\\":1,\\"column\\":11}],\\"extensions\\":{\\"code\\":\\"INTERNAL_SERVER_ERROR\\"}},{\\"message\\":\\"Variable \\\\\\"$ingredients\\\\\\" got invalid value undefined; Expected non-nullable type \\\\\\"[Ingredient_Input!]!\\\\\\" not to be null.\\",\\"locations\\":[{\\"line\\":1,\\"column\\":44}],\\"extensions\\":{\\"code\\":\\"INTERNAL_SERVER_ERROR\\"}},{\\"message\\":\\"Variable \\\\\\"$instructions\\\\\\" of required type \\\\\\"[String!]!\\\\\\" was not provided.\\",\\"locations\\":[{\\"line\\":1,\\"column\\":80}],\\"extensions\\":{\\"code\\":\\"INTERNAL_SERVER_ERROR\\"}}]}
"
`;

exports[`Mutation createRecipe Unsuccessful Wrong type of args '$ingredients' of ingredients variant 4`] = `
"{\\"errors\\":[{\\"message\\":\\"Variable \\\\\\"$title\\\\\\" of required type \\\\\\"String!\\\\\\" was not provided.\\",\\"locations\\":[{\\"line\\":1,\\"column\\":11}],\\"extensions\\":{\\"code\\":\\"INTERNAL_SERVER_ERROR\\"}},{\\"message\\":\\"Variable \\\\\\"$ingredients\\\\\\" got invalid value undefined; Expected non-nullable type \\\\\\"[Ingredient_Input!]!\\\\\\" not to be null.\\",\\"locations\\":[{\\"line\\":1,\\"column\\":44}],\\"extensions\\":{\\"code\\":\\"INTERNAL_SERVER_ERROR\\"}},{\\"message\\":\\"Variable \\\\\\"$instructions\\\\\\" of required type \\\\\\"[String!]!\\\\\\" was not provided.\\",\\"locations\\":[{\\"line\\":1,\\"column\\":80}],\\"extensions\\":{\\"code\\":\\"INTERNAL_SERVER_ERROR\\"}}]}
"
`;

exports[`Mutation createRecipe Unsuccessful Wrong type of args '$ingredients' of ingredients variant 5`] = `
"{\\"errors\\":[{\\"message\\":\\"Variable \\\\\\"$title\\\\\\" of required type \\\\\\"String!\\\\\\" was not provided.\\",\\"locations\\":[{\\"line\\":1,\\"column\\":11}],\\"extensions\\":{\\"code\\":\\"INTERNAL_SERVER_ERROR\\"}},{\\"message\\":\\"Variable \\\\\\"$ingredients\\\\\\" got invalid value undefined; Expected non-nullable type \\\\\\"[Ingredient_Input!]!\\\\\\" not to be null.\\",\\"locations\\":[{\\"line\\":1,\\"column\\":44}],\\"extensions\\":{\\"code\\":\\"INTERNAL_SERVER_ERROR\\"}},{\\"message\\":\\"Variable \\\\\\"$instructions\\\\\\" of required type \\\\\\"[String!]!\\\\\\" was not provided.\\",\\"locations\\":[{\\"line\\":1,\\"column\\":80}],\\"extensions\\":{\\"code\\":\\"INTERNAL_SERVER_ERROR\\"}}]}
"
`;

exports[`Mutation createRecipe Unsuccessful Wrong type of args '$ingredients' of ingredients variant 6`] = `
"{\\"errors\\":[{\\"message\\":\\"Variable \\\\\\"$title\\\\\\" of required type \\\\\\"String!\\\\\\" was not provided.\\",\\"locations\\":[{\\"line\\":1,\\"column\\":11}],\\"extensions\\":{\\"code\\":\\"INTERNAL_SERVER_ERROR\\"}},{\\"message\\":\\"Variable \\\\\\"$ingredients\\\\\\" got invalid value undefined; Expected non-nullable type \\\\\\"[Ingredient_Input!]!\\\\\\" not to be null.\\",\\"locations\\":[{\\"line\\":1,\\"column\\":44}],\\"extensions\\":{\\"code\\":\\"INTERNAL_SERVER_ERROR\\"}},{\\"message\\":\\"Variable \\\\\\"$instructions\\\\\\" of required type \\\\\\"[String!]!\\\\\\" was not provided.\\",\\"locations\\":[{\\"line\\":1,\\"column\\":80}],\\"extensions\\":{\\"code\\":\\"INTERNAL_SERVER_ERROR\\"}}]}
"
`;

exports[`Mutation deleteRecipeById Unsuccessful Missing args 1`] = `
"{\\"errors\\":[{\\"message\\":\\"Variable \\\\\\"$recipeUUID\\\\\\" of required type \\\\\\"String!\\\\\\" was not provided.\\",\\"locations\\":[{\\"line\\":1,\\"column\\":11}],\\"extensions\\":{\\"code\\":\\"INTERNAL_SERVER_ERROR\\"}}]}
"
`;

exports[`Mutation deleteRecipeById Unsuccessful Missing recipe 1`] = `
Object {
  "data": Object {
    "deleteRecipeById": Object {
      "errors": Array [
        Object {
          "message": "recipe didnt exist",
          "path": "recipe",
        },
      ],
      "ok": false,
      "recipe": null,
    },
  },
}
`;

exports[`Mutation updateRecipeById Unsuccessful Missing args 1`] = `
"{\\"errors\\":[{\\"message\\":\\"Variable \\\\\\"$recipeUUID\\\\\\" of required type \\\\\\"String!\\\\\\" was not provided.\\",\\"locations\\":[{\\"line\\":1,\\"column\\":11}],\\"extensions\\":{\\"code\\":\\"INTERNAL_SERVER_ERROR\\"}}]}
"
`;

exports[`Query getRecipeById Unsuccessful Missing args 1`] = `
"{\\"errors\\":[{\\"message\\":\\"Variable \\\\\\"$recipeUUID\\\\\\" of required type \\\\\\"String!\\\\\\" was not provided.\\",\\"locations\\":[{\\"line\\":1,\\"column\\":8}],\\"extensions\\":{\\"code\\":\\"INTERNAL_SERVER_ERROR\\"}}]}
"
`;

exports[`Query getRecipeById Unsuccessful Wrong type of args '$recipeUUID' of recipeUUID variant 1`] = `
"{\\"errors\\":[{\\"message\\":\\"Variable \\\\\\"$title\\\\\\" of required type \\\\\\"String!\\\\\\" was not provided.\\",\\"locations\\":[{\\"line\\":1,\\"column\\":11}],\\"extensions\\":{\\"code\\":\\"INTERNAL_SERVER_ERROR\\"}},{\\"message\\":\\"Variable \\\\\\"$ingredients\\\\\\" of required type \\\\\\"[Ingredient_Input!]!\\\\\\" was not provided.\\",\\"locations\\":[{\\"line\\":1,\\"column\\":44}],\\"extensions\\":{\\"code\\":\\"INTERNAL_SERVER_ERROR\\"}},{\\"message\\":\\"Variable \\\\\\"$instructions\\\\\\" of required type \\\\\\"[String!]!\\\\\\" was not provided.\\",\\"locations\\":[{\\"line\\":1,\\"column\\":80}],\\"extensions\\":{\\"code\\":\\"INTERNAL_SERVER_ERROR\\"}}]}
"
`;

exports[`Query getRecipeById Unsuccessful Wrong type of args '$recipeUUID' of recipeUUID variant 2`] = `
"{\\"errors\\":[{\\"message\\":\\"Variable \\\\\\"$title\\\\\\" of required type \\\\\\"String!\\\\\\" was not provided.\\",\\"locations\\":[{\\"line\\":1,\\"column\\":11}],\\"extensions\\":{\\"code\\":\\"INTERNAL_SERVER_ERROR\\"}},{\\"message\\":\\"Variable \\\\\\"$ingredients\\\\\\" of required type \\\\\\"[Ingredient_Input!]!\\\\\\" was not provided.\\",\\"locations\\":[{\\"line\\":1,\\"column\\":44}],\\"extensions\\":{\\"code\\":\\"INTERNAL_SERVER_ERROR\\"}},{\\"message\\":\\"Variable \\\\\\"$instructions\\\\\\" of required type \\\\\\"[String!]!\\\\\\" was not provided.\\",\\"locations\\":[{\\"line\\":1,\\"column\\":80}],\\"extensions\\":{\\"code\\":\\"INTERNAL_SERVER_ERROR\\"}}]}
"
`;

exports[`Query getRecipeById Unsuccessful Wrong type of args '$recipeUUID' of recipeUUID variant 3`] = `
"{\\"errors\\":[{\\"message\\":\\"Variable \\\\\\"$title\\\\\\" of required type \\\\\\"String!\\\\\\" was not provided.\\",\\"locations\\":[{\\"line\\":1,\\"column\\":11}],\\"extensions\\":{\\"code\\":\\"INTERNAL_SERVER_ERROR\\"}},{\\"message\\":\\"Variable \\\\\\"$ingredients\\\\\\" of required type \\\\\\"[Ingredient_Input!]!\\\\\\" was not provided.\\",\\"locations\\":[{\\"line\\":1,\\"column\\":44}],\\"extensions\\":{\\"code\\":\\"INTERNAL_SERVER_ERROR\\"}},{\\"message\\":\\"Variable \\\\\\"$instructions\\\\\\" of required type \\\\\\"[String!]!\\\\\\" was not provided.\\",\\"locations\\":[{\\"line\\":1,\\"column\\":80}],\\"extensions\\":{\\"code\\":\\"INTERNAL_SERVER_ERROR\\"}}]}
"
`;

exports[`Query getRecipeById Unsuccessful Wrong type of args '$recipeUUID' of recipeUUID variant 4`] = `
"{\\"errors\\":[{\\"message\\":\\"Variable \\\\\\"$title\\\\\\" of required type \\\\\\"String!\\\\\\" was not provided.\\",\\"locations\\":[{\\"line\\":1,\\"column\\":11}],\\"extensions\\":{\\"code\\":\\"INTERNAL_SERVER_ERROR\\"}},{\\"message\\":\\"Variable \\\\\\"$ingredients\\\\\\" of required type \\\\\\"[Ingredient_Input!]!\\\\\\" was not provided.\\",\\"locations\\":[{\\"line\\":1,\\"column\\":44}],\\"extensions\\":{\\"code\\":\\"INTERNAL_SERVER_ERROR\\"}},{\\"message\\":\\"Variable \\\\\\"$instructions\\\\\\" of required type \\\\\\"[String!]!\\\\\\" was not provided.\\",\\"locations\\":[{\\"line\\":1,\\"column\\":80}],\\"extensions\\":{\\"code\\":\\"INTERNAL_SERVER_ERROR\\"}}]}
"
`;

exports[`Query getRecipeById Unsuccessful Wrong type of args '$recipeUUID' of recipeUUID variant 5`] = `
"{\\"errors\\":[{\\"message\\":\\"Variable \\\\\\"$title\\\\\\" of required type \\\\\\"String!\\\\\\" was not provided.\\",\\"locations\\":[{\\"line\\":1,\\"column\\":11}],\\"extensions\\":{\\"code\\":\\"INTERNAL_SERVER_ERROR\\"}},{\\"message\\":\\"Variable \\\\\\"$ingredients\\\\\\" of required type \\\\\\"[Ingredient_Input!]!\\\\\\" was not provided.\\",\\"locations\\":[{\\"line\\":1,\\"column\\":44}],\\"extensions\\":{\\"code\\":\\"INTERNAL_SERVER_ERROR\\"}},{\\"message\\":\\"Variable \\\\\\"$instructions\\\\\\" of required type \\\\\\"[String!]!\\\\\\" was not provided.\\",\\"locations\\":[{\\"line\\":1,\\"column\\":80}],\\"extensions\\":{\\"code\\":\\"INTERNAL_SERVER_ERROR\\"}}]}
"
`;

exports[`Query getRecipeById Unsuccessful Wrong type of args '$recipeUUID' of recipeUUID variant 6`] = `
"{\\"errors\\":[{\\"message\\":\\"Variable \\\\\\"$title\\\\\\" of required type \\\\\\"String!\\\\\\" was not provided.\\",\\"locations\\":[{\\"line\\":1,\\"column\\":11}],\\"extensions\\":{\\"code\\":\\"INTERNAL_SERVER_ERROR\\"}},{\\"message\\":\\"Variable \\\\\\"$ingredients\\\\\\" of required type \\\\\\"[Ingredient_Input!]!\\\\\\" was not provided.\\",\\"locations\\":[{\\"line\\":1,\\"column\\":44}],\\"extensions\\":{\\"code\\":\\"INTERNAL_SERVER_ERROR\\"}},{\\"message\\":\\"Variable \\\\\\"$instructions\\\\\\" of required type \\\\\\"[String!]!\\\\\\" was not provided.\\",\\"locations\\":[{\\"line\\":1,\\"column\\":80}],\\"extensions\\":{\\"code\\":\\"INTERNAL_SERVER_ERROR\\"}}]}
"
`;
